lib/trento/application/integration/discovery/policies/host_policy.ex

```diff
@@ -113,6 +84,49 @@ defmodule Trento.Integration.Discovery.HostPolicy do
            os_version: os_version
          })
 
+  defp build_update_provider_command(agent_id, %CloudDiscoveryPayload{
+         provider: provider,
+         metadata: metadata
+       }) do
+    UpdateProvider.new(%{
+      host_id: agent_id,
+      provider: provider,
+      provider_data:
+        case metadata do
+          %{
+            compute: %{
```

Comment:
could we have a function here instead of a case?

Reply:
I would have something like:
```
defp build_update_provider_command(agent_id, %CloudDiscoveryPayload{
         provider: :azure,
         metadata: metadata
       }) do
...
````

To handle the different provider overtime, instesad of the `case` option